src/GafferArnold/IECoreArnoldPreview/Renderer.cpp

```diff
@@ -1416,6 +1431,19 @@ class ArnoldLight : public ArnoldObject
 				}
 			}
 
+			// Deal with light filters.
+
+			std::vector<AtNode*> lightFilterNodesVector;
+			std::vector<ArnoldShaderPtr> lightFilterShaders = arnoldAttributes->lightFilterShaders();
+			lightFilterNodesVector.reserve( lightFilterShaders.size() );
```

Comment:
I don't think we actually need this temporary vector, because we could use [`AiArrayAllocate`](https://trac.solidangle.com/arnoldpedia/chrome/site/Arnold-4.0.14.0/doc/api/group__ai__array.html#ga26d7ad1cd1775d85517d017115a74f49) to allocate the array and then use `AiArraySetPtr()` to fill it as we iterate. Best to avoid temporary allocations if possible...